src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -246,6 +243,38 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
             return project;
         }
 
+        internal static string NormalizeProjectDirectoryPath(string path)
+        {
+            if (File.Exists(path) &&
+                string.Equals(Path.GetFileName(path), Project.FileName, StringComparison.OrdinalIgnoreCase))
+            {
+                string directoryName = Path.GetDirectoryName(path);
+                if (string.IsNullOrEmpty(directoryName))
+                {
+                    directoryName = Directory.GetCurrentDirectory();
+                }
+
+                return Path.GetFullPath(directoryName);
+            }
+            else if (Directory.Exists(path) &&
+                     File.Exists(Path.Combine(path, Project.FileName)))
+            {
+                return Path.GetFullPath(path);
+            }
+
+            return null;
```

Comment:
Why? Do you want to do the opposite and convince me that null is better? because it seems to me that return string.Empty; is a very small chance that should cost literally nothing do make.